dts/arm/st/l5/stm32l5.dtsi

```diff
@@ -601,8 +601,8 @@
 			interrupts = <37 0>;
 			status = "disabled";
 			#io-channel-cells = <1>;
-			has-temp-channel;
-			has-vref-channel;
+			temp-channel = <17>;
+			vref-channel = <0>;
```

Comment:
```suggestion
			vref-channel = <0>;
			temp-channel = <17>;
```

Reply:
1. I chose to always sort by temp/vref/vbat in this order to keep consistency, but I suppose both sortings have their merits
2. In this PR I only tried to simplify and replace (and sometimes correct) what was already there. Adding missing temp/vref/vbat can be done later in another PR.